plasmapy/utils/decorators/checks.py

```diff
@@ -14,7 +14,257 @@
+                params_str = ", ".join(missing_params)
+                raise TypeError(
+                    f"Call to {self.f.__name__} is missing validated "
+                    f"params {params_str}")
+
+            # Review checks and check argument
+            for arg_name in checks:
+                self._check_value(given_args[arg_name],
+                                  arg_name,
+                                  **checks[arg_name])
+
+            return f(**given_args)
+
+        return wrapper
+
+    def _get_checks(self,
+                    bound_args: inspect.BoundArguments) -> Dict[str, Dict[str, bool]]:
+        """
+        Review function bound arguments and :attr:`checks` to build a complete 'checks'
+        dictionary. Any unspecified check key is filled with a default value.
+
+        Parameters
+        ----------
+        bound_args
+            Bound Arguments passed to the function being wrapped.
+
+        Returns
+        -------
+        Dict[str, Dict[str, bool]]
+            A complete 'checks' dictionary for checking function input arguments.
+
+        """
+        # initialize validation dictionary
+        out_checks = {}
+
+        # Iterate through parameters, determine validation keys, and build checks
+        # dictionary `out_checks`
+        for param in bound_args.signature.parameters.values():
+            # variable arguments (*args, **kwargs) not checked
+            if param.kind in (inspect.Parameter.VAR_KEYWORD,
+                              inspect.Parameter.VAR_POSITIONAL):
+                continue
+
+            # catch the (never triggered) case where bind relied on a default value
+            if param.name not in bound_args.arguments \
+                    and param.default is not param.empty:
+                bound_args.arguments[param.name] = param.default
+
+            # grab the checks dictionary for the desired parameter
+            try:
+                param_in_checks = self.checks[param.name]
+            except KeyError:
+                # checks for parameter not specified
+                continue
+
+            # build `out_checks`
+            # read checks and/or apply defaults values
+            out_checks[param.name] = {}
+            for v_name, v_default in self._check_item_defaults.items():
+                out_checks[param.name][v_name] = \
+                    param_in_checks.get(v_name, v_default)
+
+        return out_checks
+
+    @property
+    def _check_item_defaults(self) -> Dict[str, bool]:
+        """Default values for the possible 'check' keys."""
+        # adding a new validation key and default value here will automatically
+        # update the checks of the `checks` passed in
+        _defaults = {
+            'can_be_negative': True,
+            'can_be_complex': False,
+            'can_be_inf': True,
+            'can_be_nan': True,
+            'none_shall_pass': False,
+        }
+        return _defaults
```

Comment:
Hm, since these are static, wouldn't it be enough to have this as a class field, as in the following?
```python
class CheckValues:
    _defaults = {
        'can_be_negative': True,
        'can_be_complex': False,
        'can_be_inf': True,
        'can_be_nan': True,
        'none_shall_pass': False,
    }
```

Reply:
Unless the whole point is having the docstring attached to this dictionary, if that's the case - I'm not sure how else to do this.

Reply:
@StanczakDominik If defined as a class attribute, then a user could potentially alter the default values. That is,

```python
>>> from plasmapy.utils.decorators import CheckValues
>>> cv = CheckValues()
>>> cv._defaults
{'can_be_negative': True,
 'can_be_complex': False,
 'can_be_inf': True,
 'can_be_nan': True,
 'none_shall_pass': False}
>>>
>>> cv._defaults['none_shall_pass'] = True
>>> cv._defaults
{'can_be_negative': True,
 'can_be_complex': False,
 'can_be_inf': True,
 'can_be_nan': True,
 'none_shall_pass': True}
```

By declaring it as a property, I can then prevent any unwanted changes to the default values.  Not that this situation is likely to come up, if ever.

Reply:
Oh, I wouldn't have thought of that. Since defaults are marked as implementation details by `_` prefixed, I think any user who bothers modifying them either knows what they're doing or is unlikely to exist in the first place, as you said.

Reply:
You're right, this way is not a very pythonic approach, and probably does not deserve this level of rigidity.  I'm continuously trying to break my old-school rigid programming ways and rely more on python's conventions.

I'll move the defaults into a "protected" class attribute.